profiler/src/main/java/com/splunk/opentelemetry/logs/BatchingLogsProcessor.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.logs;
+
+import static java.util.Objects.requireNonNull;
+
+import com.splunk.opentelemetry.profiler.util.HelpfulExecutors;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * A log processor that batches logs until a max number are buffered or a time limit has passed,
+ * which ever comes first.
+ */
+public class BatchingLogsProcessor implements LogsProcessor {
+
+  private static final int DEFAULT_MAX_BATCH_SIZE = 250;
+  private static final Duration DEFAULT_MAX_TIME_BETWEEN_BATCHES = Duration.ofSeconds(10);
+
+  private final int maxBatchSize;
+  private final Duration maxTimeBetweenBatches;
+  private final List<LogEntry> batch;
+  private final LogsExporter exporter;
+  private final ScheduledExecutorService executorService;
+  private final Object lock = new Object();
+  private WatchdogTimer watchdog;
+
+  public BatchingLogsProcessor(Builder builder) {
+    this.maxTimeBetweenBatches = builder.maxTimeBetweenBatches;
+    this.maxBatchSize = builder.maxBatchSize;
+    this.exporter = builder.batchAction;
+    this.executorService = builder.executorService;
+    batch = new ArrayList<>(maxBatchSize);
+  }
+
+  public void start() {
+    synchronized (lock) {
+      if (watchdog != null) {
+        throw new IllegalStateException("Already running");
+      }
+      watchdog = new WatchdogTimer(maxTimeBetweenBatches, this::syncDoExport, executorService);
+      watchdog.start();
+    }
+  }
+
+  void stop() {
+    synchronized (lock) {
+      if (watchdog == null) {
+        throw new IllegalStateException("Not running");
```

Comment:
if it's null, could we just no-op this, rather than throwing an exception? Seems like it shouldn't really be harmful.

Reply:
It's fair, but I just wanted to keep the state correct, and the exception seemed like a way to do that while keeping parity with the `start()` state check.